src/two_finger_grasp_generator.cpp

```diff
@@ -0,0 +1,917 @@
+  t = (height / 2.0 - point_a[2]) / (point_b[2] - point_a[2]);  // parameterization of line segment in 3d
+  if (intersectionHelper(t, point_a[0], point_a[1], point_b[0], point_b[1], depth, width, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= u;
+    //   intersection[1]= v;
+    //   intersection[2]= height / 2.0;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::BLUE, 0.005);
+    // }
+    return true;
+  }
+
+  t = (-height / 2.0 - point_a[2]) / (point_b[2] - point_a[2]);
+  if (intersectionHelper(t, point_a[0], point_a[1], point_b[0], point_b[1], depth, width, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= u;
+    //   intersection[1]= v;
+    //   intersection[2]= -height / 2.0;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::CYAN, 0.005);
+    // }
+    return true;
+  }
+
+  // check if line segment intersects XZ faces of cuboid (y = +/- width/2)
+  t = (width / 2.0 - point_a[1]) / (point_b[1] - point_a[1]);
+  if (intersectionHelper(t, point_a[0], point_a[2], point_b[0], point_b[2], depth, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= u;
+    //   intersection[1]= width / 2.0;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::GREEN, 0.005);
+    // }
+    return true;
+  }
+
+  t = (-width / 2.0 - point_a[1]) / (point_b[1] - point_a[1]);
+  if (intersectionHelper(t, point_a[0], point_a[2], point_b[0], point_b[2], depth, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= u;
+    //   intersection[1]= -width / 2.0;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::LIME_GREEN, 0.005);
+    // }
+    return true;
+  }
+
+  // check if line segment intersects YZ faces of cuboid (x = +/- depth/2)
+  t = (depth / 2.0 - point_a[0]) / (point_b[0] - point_a[0]);
+  if (intersectionHelper(t, point_a[1], point_a[2], point_b[1], point_b[2], width, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= depth / 2.0;
+    //   intersection[1]= u;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::RED, 0.005);
+    // }
+    return true;
+  }
+
+  t = (-depth / 2.0 - point_a[0]) / (point_b[0] - point_a[0]);
+  if (intersectionHelper(t, point_a[1], point_a[2], point_b[1], point_b[2], width, height, u, v))
+  {
+    // if (verbose_)
+    // {
+    //   intersection[0]= -depth / 2.0;
+    //   intersection[1]= u;
+    //   intersection[2]= v;
+    //   visual_tools_->publishSphere(intersection, rviz_visual_tools::PINK, 0.005);
+    // }
+    return true;
+  }
+
+  // no intersection found
+  return false;
+}
+
+bool TwoFingerGraspGenerator::intersectionHelper(double t, double u1, double v1, double u2, double v2, double a,
+                                                 double b, double& u, double& v)
```

Comment:
nit: const all inputs to make it more obvious u and v are outputs, also variable names?

Reply:
legacy code